commands.go

```diff
@@ -110,7 +110,16 @@ var Commands = []cli.Command{
 
 			host, err := store.Create(name, driver, c)
 			if err != nil {
-				log.Fatal(err)
+				log.Errorf("Error creating host: %s", err)
+				if c.GlobalBool("debug") {
+					log.Fatal(err)
+				} else {
+					log.Warnf("Removing provisioned machine.  Use --debug to prevent removal.")
+					// we know there was an error so do not check for the error on removal
+					// instead we will Fatal with a message to prevent spamming with error messages
+					store.Remove(name, true)
```

Comment:
The way things are it almost always will fail.  That is the reason I chose to just do it and then show the final error.  Otherwise you get multiple fails - which may be ok just a little confusing.